default.nix

```diff
@@ -8,7 +8,7 @@
     (builtins.fetchTarball
       "https://github.com/input-output-hk/iohk-nix/archive/7f57f750e27c84def8e0ed2492eea01ea957cbf2.tar.gz")
     { }
-, nixpkgsSrc ? haskellNix.sources.nixpkgs-2009
+, nixpkgsSrc ? haskellNix.sources.nixpkgs-2105
```

Comment:
This one is very heavyweight, but:
- is required for macOS to get the fix for https://github.com/NixOS/nixpkgs/issues/91748.
- matches the default value in `shell.nix`!

Reply:
This comes with two further questions:
1. If each bump is so heavyweight, should we just jump to `2111` right in this PR?
2. Would you take a PR that attempts to trim down unused dependencies, packages, and improve Nix here and there to improve build time? I find it a fitting next step in my current journey.

Reply:
Yes, let's bump to `21.11` if a newer nixpkgs is needed.

Reply:
Sure, I'm on it.